Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/cut_with_plane.h

```diff
@@ -0,0 +1,649 @@
+// Copyright (c) 2024 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Sébastien Loriot
+
+
+#ifndef CGAL_POLYGON_MESH_PROCESSING_CUT_WITH_PLANE_H
+#define CGAL_POLYGON_MESH_PROCESSING_CUT_WITH_PLANE_H
+
+#include <CGAL/license/Polygon_mesh_processing/corefinement.h>
+
+#include <CGAL/Named_function_parameters.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+#include <CGAL/Polygon_mesh_processing/connected_components.h>
+#include <CGAL/Polygon_mesh_processing/border.h>
+#ifndef CGAL_PLANE_CLIP_DO_NOT_USE_TRIANGULATION
+#include <CGAL/Polygon_mesh_processing/triangulate_faces.h>
+#endif
+#ifndef CGAL_PLANE_CLIP_DO_NOT_USE_BOX_INTERSECTION_D
+#include <CGAL/Polygon_mesh_processing/self_intersections.h>
+#endif
+
+namespace CGAL {
+namespace Polygon_mesh_processing {
+
+template <class PolygonMesh>
+struct Default_cut_visitor
+{
+  void before_edge_split(typename boost::graph_traits<PolygonMesh>::halfedge_descriptor, PolygonMesh&) {}
+  void edge_split(typename boost::graph_traits<PolygonMesh>::halfedge_descriptor, PolygonMesh&) {}
+  void vertices_on_cut(const std::vector<typename boost::graph_traits<PolygonMesh>::vertex_descriptor>&, PolygonMesh&){}
+};
+
+// TODO: doc me or hide me in the np
+template <class Kernel>
+struct Orthogonal_cut_plane_traits
+{
+  using FT = typename Kernel::FT;
+  using Plane_3 = std::pair<int, FT>;
+  using Point_3 = typename Kernel::Point_3;
+
+  struct Oriented_side_3
+  {
+    Oriented_side operator()(const Plane_3& plane, const Point_3& p)  const
+    {
+      if (p[plane.first]==plane.second) return ON_ORIENTED_BOUNDARY;
+      return p[plane.first]<plane.second ? ON_NEGATIVE_SIDE : ON_POSITIVE_SIDE;
+    }
+  };
+
+  // TODO here we should integrate the epsilon to reuse existing points (moving them on the grid as an option?)
```

Comment:
actually we might want to move input points in the plane and update the oriented side.